engine_test.go

```diff
@@ -156,6 +156,7 @@ input:
       id: RootObject
       properties:
         name:
+          default: not
           type:
             type_id: string
```

Comment:
yes TestE2E

Reply:
The code I'm citing here _is_ `TestE2E()`.

I think that there are three cases which ought to be tested:
1. no default specified in the workflow; value specified in the invocation (because it has to be)
2. default specified in the workflow; no value specified in the invocation (so the default value is used)
3. default specified in the workflow; value specified in the invocation (so the default value is not used)

I think we have the third case covered here; I think the second is covered in `TestE2EWorkflowDefaultInput()`.  So, the question is, do we need an explicit test for the first case, or do we think that that's the "normal" case which is adequately exercised in the rest of the testing?